addons/mail/models/discuss/discuss_channel.py

```diff
@@ -958,25 +959,21 @@ def _channel_seen(self, last_message_id=None, allow_older=False):
         if last_message_id is not False and not last_message:
             return
         self._set_last_seen_message(last_message, allow_older=allow_older)
-        data = {
-            'channel_id': self.id,
-            'last_message_id': last_message.id,
-            'partner_id': self.env.user.partner_id.id,
-        }
-        target = self if self.channel_type == 'chat' else self.env.user.partner_id
-        self.env['bus.bus']._sendone(target, 'discuss.channel.member/seen', data)
         return last_message.id
 
     def _set_last_seen_message(self, last_message, allow_older=False):
         """
-        Set last seen message of `self` channels for the current user.
+        Set last seen message of `self` channels for the current persona.
         :param last_message: the message to set as last seen message
         :param allow_order: whether to allow setting and older message
         as the last seen message.
         """
+        current_partner, current_guest = self.env["res.partner"]._get_current_persona()
+        if not current_partner and not current_guest:
+            return
```

Comment:
Is it actually possible to reach this code without having either of them?

Reply:
I think so, `_get_current_personna` return an empty recordset for `current_partner` when the user is public. Whatsapp users are public but do not use guests hence this situation. Moreover, the following code should not be executed for public partners so the previous code was not correct.